lib/active_support/cache/dual_rails_store.rb

```diff
@@ -0,0 +1,71 @@
+require 'active_support/cache'
+require 'active_support/version'
+
+module ActiveSupport
```

Comment:
I would add all the explanation you added in the PR as a comment here, because even if this is temporary I think having an explanation in the code that survives the PR life will make it easier for people that come across to this file to understand what it's supposed to do.

I would also mention it's adapted from <https://github.com/mezis/level2>